include/swift/AST/DiagnosticsSema.def

```diff
@@ -3216,6 +3221,18 @@ ERROR(trailing_closure_requires_parens,none,
       "trailing closure requires parentheses for disambiguation in this"
       " context", ())
 
+ERROR(opaque_type_no_underlying_type_candidates,none,
+      "function declares an opaque return type, but has no return statements "
+      "in its body from which to infer an underlying type", ())
```

Comment:
This should be allowed IMHO:

```
func f() -> __opaque P { fatalError() }
```

Once you have resilience working here, you can compile the metadata accessor as 'unreachable' and never concretize the type in SILGen

Reply:
It would be easier to support unreachable opaque types with the runtime support in place, because we could then leave the underlying type opaque during SILGen/IRGen since the underlying type is never reachable.